src/util.h

```diff
@@ -69,15 +70,50 @@ inline std::string _(const char* psz)
 void SetupEnvironment();
 bool SetupNetworking();
 
+enum LogFlags : uint32_t {
+    LOG_NONE        = 0,
+    LOG_NET         = (1 <<  0),
+    LOG_TOR         = (1 <<  1),
+    LOG_MEMPOOL     = (1 <<  2),
+    LOG_HTTP        = (1 <<  3),
+    LOG_BENCH       = (1 <<  4),
+    LOG_ZMQ         = (1 <<  5),
+    LOG_DB          = (1 <<  6),
+    LOG_RPC         = (1 <<  7),
+    LOG_ESTIMATEFEE = (1 <<  8),
+    LOG_ADDRMAN     = (1 <<  9),
+    LOG_SELECTCOINS = (1 << 10),
+    LOG_REINDEX     = (1 << 11),
+    LOG_CMPCTBLOCK  = (1 << 12),
+    LOG_RAND        = (1 << 13),
+    LOG_PRUNE       = (1 << 14),
+    LOG_PROXY       = (1 << 15),
+    LOG_MEMPOOLREJ  = (1 << 16),
+    LOG_LIBEVENT    = (1 << 17),
+    LOG_COINDB      = (1 << 18),
+    LOG_QT          = (1 << 19),
+    LOG_ALL         = ~(uint32_t)0,
+};
+
 /** Return true if log accepts specified category */
-bool LogAcceptCategory(const char* category);
+static inline bool LogAcceptCategory(uint32_t category)
+{
+    return (logCategories.load(std::memory_order_relaxed) & category) != 0;
+}
+
+/** Returns a string with the supported log categories */
+std::string ListLogCategories();
+
+/** Return true of str parses as a log category and set the flags in f */
+bool ParseLogCategory(uint32_t *f, const std::string *str);
+
 /** Send a string to the log output */
 int LogPrintStr(const std::string &str);
 
-#define LogPrintf(...) LogPrint(NULL, __VA_ARGS__)
+#define LogPrintf(...) LogPrintStr(tfm::format(__VA_ARGS__))
 
 template<typename... Args>
-static inline int LogPrint(const char* category, const char* fmt, const Args&... args)
+static inline int LogPrint(uint32_t category, const char* fmt, const Args&... args)
```

Comment:
Yes, that would add some type safety. It would make it more of a hassle to submit a message to more categories at once but that isn't supported now either.